ludwig/data/dataframe/dask.py

```diff
@@ -0,0 +1,109 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2020 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import multiprocessing
+import os
+
+import dask
+import dask.array as da
+import dask.dataframe as dd
+
+from ludwig.constants import NAME, PROC_COLUMN
+from ludwig.data.dataset.parquet import ParquetDataset
+from ludwig.data.dataframe.base import DataFrameEngine
+from ludwig.utils.data_utils import DATA_PROCESSED_CACHE_DIR, DATASET_SPLIT_URL
+from ludwig.utils.misc_utils import get_combined_features
+
+TMP_COLUMN = '__TMP_COLUMN__'
+
+
+def set_scheduler(scheduler):
+    dask.config.set(scheduler=scheduler)
+
+
+class DaskEngine(DataFrameEngine):
+    def __init__(self, parallelism=None):
+        self._parallelism = parallelism or multiprocessing.cpu_count()
+
+    def empty_df_like(self, df):
+        # Our goal is to preserve the index of the input dataframe but to drop
+        # all its columns. Because to_frame() creates a column from the index,
+        # we need to drop it immediately following creation.
+        return df.index.to_frame(name=TMP_COLUMN).drop(columns=[TMP_COLUMN])
+
+    def parallelize(self, data):
+        return data.repartition(self.parallelism)
+
+    def persist(self, data):
+        return data.persist()
```

Comment:
I should point out that, contrary to Dask Distributed's `client.persist()` implementation which returns a Dask DataFrame containing references to futures, Dask-on-Ray currently uses plain Dask's default behavior of returning a Dask DataFrame containing the fully materialized data. We have an [open issue](https://github.com/ray-project/ray/issues/14227) for addressing this in the next sprint or two, but because of Dask's `df.persist()` implementation, the API might have to be slightly different (e.g. a `ray_persist()` function that we provide).

Reply:
Good news here! Proper `dask.persist()` support for the Dask-on-Ray scheduler has been implemented, and should just work as expected: https://github.com/ray-project/ray/pull/14294

Reply:
Thanks! Adding a reference to this issue. Looks like it's closed, what the resolution to fully support the expected semantics of `df.persist()`? Any timeline on when this will be released?

Reply:
Looks like my comment came a little late, I will remove the link to the ticket then. This is great, thanks!